### DISCUSSION TITLE
Is there support for 3 ROM chip SNES cartridges?

### DISCUSSION THREAD

**Discussion by mikegabriel** [sanni/cartreader#119] [Need help]
Hi there,

I've got an uncommon PCB that came to me as not working.  It's a copy of Super Metroid but it's a 3 ROM PCB ([SHVC-3A3M-01](https://snescentral.com/pcbboards.php?chip=SHVC-3A3M-01)) which in my research has concluded it's just an uncommon board that wasn't used a ton.  And I found one other forum post with corrupted MaskROMS on this specific board.

Anyways, onto the problem...  the game itself doesn't boot, the SNES detects it but just gives a black screen.  I generally like to try and dump the ROM off just to see and for this specific game, the CRC failed.  I noticed upon closer inspection, the cartridge reader is picking it up as `Chips: 2`.

So I guess what I am wondering if this is just an unsupported board and if you could confirm that the cart reader programming does or doesn't handle 3 ROM PCBs for SNES?

![IMG_6120](https://user-images.githubusercontent.com/481907/113899224-d2548400-9789-11eb-968d-4fb98347ea75.jpeg)

![20200917_052312_2773_pcb_front](https://user-images.githubusercontent.com/481907/113900921-91f60580-978b-11eb-8946-f8ee4d3aef3c.jpeg)

Cheers

**Comment by sanni** [comment]
The Cart Reader has no way of actually counting the maskroms inside the cartridge. The number you see is just read out of the ROM header and will be the same for this game no matter if the PCB has one, two or three maskroms. I had removed the Chips entry in the past since I don't find it useful but somehow it snuck itself back in apparently. It's basically a duplicate of the ICs entry.
Here is what the value means:
0 = ROM only, 1 = ROM & RAM, 2 = ROM & Save RAM,  3 = ROM & DSP1, 4 = ROM & RAM & DSP1, 5 = ROM & Save RAM & DSP1, 19 = ROM & SFX, 227 = ROM & RAM & GameBoy data, 243 = CX4, 246 = ROM & DSP2

I had a 3 maskrom Final Fantasy 6 and it read-out just like a normal game. So to the Cart Reader there is no difference since it's all handled by the MAD-1 and 74LS139 chip inside the game cartridge.

To troubleshoot your cartridge you need to dump it and then compare the dump with a known good ROM of the same game using a hex editor like Nexus HxD and the built-in compare function.  The offsets at which the dump differs from the ROM should make a pattern that will give you the address lines that are bad which will then lead you to the trace that will have to be repaired or maskrom chip that needs to be resoldered. It could also be the MAD-1 chip failing to switch between SRAM and maskroms which would give you seemingly random errors in the dump.

**Reply by mikegabriel** [reply]
Cool, I was looking at the SNES code and found those values but figured easier to ask for clarification.

Yeah, I dumped the ROM and compared, using the compare function was basically like 30,000 differences before I just closed it.   Right from the get-go I am seeing differences.

I reflowed the whole board but got no difference in pre-reflow dump and post-reflow.  Wasn't aware of the MAD-1 chip but I also reflowed it as part of the process.

Might be a dead maskrom and just make a nice display cartridge for someone.

Left is dumped cartridge, Right is a verified copy.

![Screen Shot 2021-04-08 at 7 46 55 AM](https://user-images.githubusercontent.com/481907/114038126-d4c4e580-983e-11eb-96d5-eb4fd9a3f270.png)

Cheers and thanks for the little bit of education on it!

**Comment by sanni** [comment]
Seems pretty random, here is what you can do:

Assuming there are no important saves on the SRAM you can write the attached save file to the SRAM. If the maskrom is broken the SRAM should still write fine but if the SRAM write fails too then the problem lies either in the PCB or in the MAD-1/74LS139 chip.
If the write fails dump the SRAM again and compare it in a hex editor with the original file attached here. This should make it easier to spot which data pin is bad.

[64_64K.zip (unzip before writing)](https://github.com/sanni/cartreader/files/6283843/64_64K.zip)
